tests/diff_test.sh

```diff
@@ -0,0 +1,70 @@
+# Script for diffing a certain output file of a test and printing nice output
+#
+# While we could put this code into CMakeLists.txt, cmake prints the whole
+# command that failed, which will be really long and ugly. This way, only the
+# line executing this script will show up.
+
+# Call this script with 4 parameters:
+
+# 1. name/path of diff executable
+DIFF_EXE=$1
+
+# 2. short name of the reference file: "testname/filename"
+PRETTY_FILENAME=$2
```

Comment:
It took me a while in the code below to understand that this includes the directory and the name (because `.../output-${PRETTY_FILENAME}.cmp.notime` looks like it only references the *name* of the file). Could this variable be renamed to `DIR_AND_NAME` or similar?